### PR TITLE
Tprime evaluation and Solve TICA problem with not-symmetric Correlation Matrix

### PR DESCRIPTION
**Pull request by nicolopedrani**
## Description
- [ ] definition of `tprime_evaluation` function outside `create_time_lagged_dataset` function.  
- [ ] Now it's possible to pass a **tprime** argument (optional) to `create_time_lagged_dataset`. If not passed then the rescaled time is computed from the **unbias time** and **logweights**      
- [ ] Attribute **symmetrize** added to `TICA` object. Default value set to `True`, and enforce the symmetrization of Correlation matrices. If set to `False` the `Compute_TICA` function solve the TICA problem without symmetrizing `C_tau` 

## Todos
  - check what happens if the correlation matrices are not symmetric

## Questions
- [ ] The dataset is composed by the correlated couples ( x_i(t) , x_i(t+lag) ). When the data is divided between training set and validation set (`RandomSplit`) what does it happen to the **time ordering** of this set? is the correlation matrix computed properly? the **time ordering** is not important (the correlation matrix is simply a weighted average), but dividing the whole dataset into two different datasets may lead to different estimation of both `C_tau` and `C_0`, compared to the estimation obtained by using the whole dataset.  
- [ ] what does it happend if the eigenvales and eigenfunctions are complex ? 

## Status
- [ ] Ready to go

### CODE REVIEW

**Review by luigibonati — CHANGES_REQUESTED** [review]

**Review by luigibonati — APPROVED** [review]

File: mlcvs/utils/data.py [hunk 1]
```diff
@@ -33,6 +33,46 @@ def closest_idx_torch(array, value):
...
+        # when the bias is not deposited the value of bias potential is minimum 
+        logweights -= torch.max(logweights)
+        # bug: exp(logweights/lognorm) != exp(logweights)/norm, where norm is sum_i beta V_i
+        # are we changing the temperature? 
+        """ possibilities:
+            1) logweights /= torch.min(logweights) -> logweights belong to [0,1] 
+            2) pass beta as an argument, then logweights *= beta
+            3) tprime = dt * torch.cumsum( torch.exp( torch.logsumexp(logweights,0) ) ,0)
+            4) tprime = dt *torch.exp ( torch.log (torch.cumsum (torch.exp(logweights) ) ) )
+        """
```
[the remark this answers is not in the document]
Comment (nicolopedrani) [reply mlcvs/utils/data.py:64]: do you mean instead of doing logweights -= max(logweights) ? 
I can try, but I think it will not change the estimates relative to the time correlation functions.

**Comment by luigibonati** [comment]
@nicolopedrani why did you added a new function `correlation_matrices`?

**Comment by nicolopedrani** [comment]
> @nicolopedrani why did you add a new function `correlation_matrices`? 

I will remove that. It was a test, to check if it is different to evaluate the covariances using all data. 
Tomorrow I will remove that "new" implentation
Don't work until late
